mllib/src/main/scala/org/apache/spark/ml/classification/GBTClassifier.scala

```diff
@@ -248,12 +269,38 @@ class GBTClassificationModel private[ml](
     if (prediction > 0.0) 1.0 else 0.0
   }
 
+  override protected def predictRaw(features: Vector): Vector = {
+    val treePredictions = _trees.map(_.rootNode.predictImpl(features).prediction)
+    val prediction = blas.ddot(numTrees, treePredictions, 1, _treeWeights, 1)
+    Vectors.dense(Array(-prediction, prediction))
+  }
+
+  override protected def raw2probabilityInPlace(rawPrediction: Vector): Vector = {
+    rawPrediction match {
+      // The probability can be calculated for positive result:
+      // p+(x) = 1 / (1 + e^(-2 * F(x)))
+      // and negative result:
+      // p-(x) = 1 / (1 + e^(2 * F(x)))
+      case dv: DenseVector =>
+        var i = 0
+        val size = dv.size
+        while (i < size) {
+          dv.values(i) = 1 / (1 + math.exp(-2 * dv.values(i)))
```

Comment:
my concern is that this is hard coded to logistic loss. Maybe we can add a static method to GBTClassificationModel 

````scala
private def classProbability(class: Int, loss: String, rawPrediction: Double): Double = {
  loss match {
    case "logistic" => ...
    case _ => throw new Exception("Only logistic loss is supported ...")
  }
}
````